framework/test_app/runners/cloud_run/cloud_run_xds_client_runner.py

```
@@ -0,0 +1,181 @@
+                    {
+                        "image": image_name,
+                        "ports": [
+                            {
+                                "containerPort": test_port,
+                                "name": "h2c",
+                            }
+                        ],
+                        "args": [
+                            f"--server={server_target}",
+                            "--secure_mode=true",
+                        ],
+                        "env": [
+                            {
+                                "name": "GRPC_EXPERIMENTAL_XDS_AUTHORITY_REWRITE",
+                                "value": "true",
+                            },
+                            {
+                                "name": "GRPC_EXPERIMENTAL_XDS_SYSTEM_ROOT_CERTS",
+                                "value": "true",
+                            },
+                            {
+                                "name": "GRPC_EXPERIMENTAL_XDS_GCP_AUTHENTICATION_FILTER",
+                                "value": "true",
+                            },
+                        ],
+                    }
+                ],
+                "service_mesh": {
+                    "mesh": mesh_name,
+                    "dataplaneMode": "PROXYLESS_GRPC",
+                },
+            },
+        }
+        logger.info("Deploying Cloud Run service '%s'", service_name)
+        self.cloud_run_api_manager.create_service(service_name, service_body)
+        # Allow unauthenticated requests for `LoadBalancerStatsServiceClient`
+        # and `CsdsClient` to retrieve client statistics.
+        policy_body = {
+            "policy": {
+                "bindings": [
+                    {
+                        "role": "roles/run.invoker",
+                        "members": ["allUsers"],
+                    }
+                ],
+            },
+        }
+        self.cloud_run_api_manager.setIamPolicy(service_name, policy_body)
```

Make setIamPolicy failure throw exception instead of returning False so it is propagated to the testcase and fails the test.

---

Done.